sdk/core/azure-core/src/main/java/com/azure/core/util/Header.java

```diff
@@ -82,6 +95,10 @@ public String getName() {
      * @return the value of this Header
      */
     public String getValue() {
+        if (value != null) {
+            return value;
+        }
+
         checkCachedStringValue();
         return CACHED_STRING_VALUE_UPDATER.get(this);
```

Comment:
I'm not sure this method is correct. It will only return the first value, even if new values are added. You need to ensure you're working correctly with the cached value. Same with query params below.

Reply:
`addValue` will set `value` to null if the header was created with a single value or set the cached String value to null when there are multiple values in the header, same logic for query parameter.

```java
public void addValue(String value) {
  if (values == null) {
      values = new ArrayList<>(4); // 4 was selected to add a buffer of 2 as seen in the constructor.
      values.add(this.value);
      this.value = null;
  }

  this.values.add(value);
  CACHED_STRING_VALUE_UPDATER.set(this, null);
}
```